katcp/core.py

```diff
@@ -299,7 +299,23 @@ def format_argument(self, arg):
         elif isinstance(arg, bool):
             return str(int(arg))
         else:
-            return str(arg)
+            try:
+                return str(arg)
+            except Exception:
```

Comment:
Should this rather catch only encoding errors? If it's some class that just doesn't support conversion to str at all, it seems like this is going to hide a real error.

Reply:
Yeah thats a good point. Changing to catching `UnicodeEncodeError`